src/meta/processors/zone/AddHostsProcessor.cpp

```diff
@@ -59,10 +59,11 @@ void AddHostsProcessor::process(const cpp2::AddHostsReq& req) {
 
     // Automatic generation zone
     auto zoneName = folly::stringPrintf("default_zone_%s_%d", host.host.c_str(), host.port);
-    auto zoneIdRet = getZoneId(zoneName);
-    if (nebula::ok(zoneIdRet)) {
-      LOG(ERROR) << "Zone " << zoneName << " have existed";
-      code = nebula::error(zoneIdRet);
+    auto zoneRet = zoneExist(zoneName);
+    if (zoneRet != nebula::cpp2::ErrorCode::E_ZONE_NOT_FOUND) {
+      if (zoneRet == nebula::cpp2::ErrorCode::SUCCEEDED)
+        LOG(ERROR) << "Zone " << zoneName << " have existed";
+      code = nebula::cpp2::ErrorCode::E_KEY_HAS_EXISTS;
```

Comment:
Perhaps you need to return the error code from zoneRet, otherwise, some kvstore error could be vanished. Besides, the `if` in line 64 miss a bracket